confdgnmi/src/confd_gnmi_server.py

```diff
@@ -57,6 +56,54 @@ def get_and_connect_adapter(self, username, password):
         log.debug("<== adapter=%s", adapter)
         return adapter
 
+    @staticmethod
+    def get_val_encoding(val):
+        log.debug("==> val=%s", val)
+        encoding = None
+        if val.HasField("json_val"):
+            encoding = gnmi_pb2.Encoding.JSON
+        elif val.HasField("json_ietf_val"):
+            encoding = gnmi_pb2.Encoding.JSON_IETF
+        elif val.HasField("ascii_val"):
+            encoding = gnmi_pb2.Encoding.ASCII
+        elif val.HasField("bytes_val"):
+            encoding = gnmi_pb2.Encoding.BYTES
+        else:
+            found = False
+            for a in ["string_val", "int_val", "uint_val", "bool_val", "bytes_val", "float_val", "leaflist_val"]:
+                if  val.HasField(a):
+                    found = True
+                    break
+            if found:
+                encoding = gnmi_pb2.Encoding.PROTO
```

Comment:
Note you don't need the variable `found` at all - just assign to `encoding` on line 75.